### PR TITLE
Support for edn tagged reader literal notation as noted in issue 113.

### PR DESCRIPTION
**Pull request by jconti**
Added tagged literal support as described in #113.

### CODE REVIEW

File: test/clj_time/coerce_test.clj [hunk 1]
```diff
@@ -144,6 +145,25 @@
...
+  (is (nil? (to-edn "x")))
+  (is (= (date-time 1970 1 1 0 0 0 0)
+         (edn/read-string {:readers data-readers}
+                          (to-edn 0))))
+  (are [o] (= (date-time 1998 4 25 0 0 0 0)
+               (edn/read-string {:readers data-readers}
+                                (to-edn o)))
+    (date-time 1998 4 25)
+    (date-midnight 1998 4 25)
+    (date-midnight 1998 4 25)
```
Comment (seancorfield) [inline test/clj_time/coerce_test.clj]: Should this second `date-midnight` line be some other call?

Comment (jconti) [reply test/clj_time/coerce_test.clj]: It is a paste error.  The tests are intended to be the same cases as in test-to-string.  Thank you for the catch.

**Comment by seancorfield** [comment]
Nice! I left a question on one line of the commit.

Also, could you update the README as part of this PR with some examples? (the readme examples are run as tests with Midje so they serve two purposes: documentation *and* additional tests).

Thanks.

**Comment by seancorfield** [comment]
Many thanks! 👍

**Comment by seancorfield** [comment]
BTW, the test instructions were correct as far as I can tell. If you `rm test/readme.clj` then run `lein test-all`, the readme test is regenerated and run automatically as part of `lein test-all`:
```
(! 610)-> rm test/readme.clj 

Thu Oct 11 14:36:29
(sean)-(jobs:0)-(/Developer/workspace/clj-time)
(! 611)-> lein test-all
Performing task 'test' with profile(s): 'dev,spec,base,system,user,provided,midje'
... lots of warnings snipped ...
lein test clj-time.coerce-test

lein test clj-time.core-test

lein test clj-time.format-test

lein test clj-time.inst-test

lein test clj-time.jdbc-test

lein test clj-time.local-test

lein test clj-time.periodic-test

lein test clj-time.predicates-test

lein test clj-time.spec-test

lein test clj-time.types-test

lein test readme

Ran 127 tests containing 622 assertions.
0 failures, 0 errors.
```
Can you confirm you get that behavior _without_ running `lein test-readme`

**Comment by jconti** [comment]
Yes, that is what I am getting.  I also cleaned up the readme to be slightly more concise, and more consistent with the rest of the text.

**Comment by seancorfield** [comment]
Since I already merged the PR, feel free to send any further updates in a separate PR. I'm updating the change log and plan to cut 0.14.6 later today with these additions.

**Comment by jconti** [comment]
I removed the explicit call to test-readme in the instructions.  If I run it I get:
```
> lein test-readme
'test-readme' is not a task. See 'lein help'.
```
But a plain `lein test` picks up `readme.clj` and runs it.

**Comment by jconti** [comment]
Ok, I will follow up with another.  Sorry for the extra work.  I need github to do a convincing preview of the markdown for me.  Much appreciated!

**Pull request merged**